src/NewProject/source_grid.cpp

```diff
@@ -132,16 +132,22 @@ void sourceGrid::AddDirectories() {
   QString pathName = QFileDialog::getExistingDirectory(
       this, tr("Select Directory"), "",
       QFileDialog::ShowDirsOnly | QFileDialog::DontResolveSymlinks);
-  if ("" == pathName) return;
-  QString folder =
-      pathName.right(pathName.size() - (pathName.lastIndexOf("/") + 1));
-  QString path = pathName.left(pathName.lastIndexOf("/"));
-
-  filedata fdata;
-  fdata.m_isFolder = true;
-  fdata.m_fileName = folder;
-  fdata.m_filePath = path;
-  AddTableItem(fdata);
+  QDir findFiles = pathName;
+  QStringList files = findFiles.entryList(QStringList() << "*.v"
+                                                        << "*.sv"
+                                                        << "*.vhd",
+                                          QDir::Files);
+  foreach (QString str, files) {
+    QString folder = str.right(str.size() - (str.lastIndexOf("/") + 1));
```

Comment:
this won't work on windows. please use QFileInfo instead. and suffix as well.

Reply:
Fixed.